src/WebJobs.Extensions.DurableTask.Analyzers/Analyzers/ActivityFunction/FunctionAnalyzer.cs

```diff
@@ -128,27 +142,92 @@ private bool TryGetFunctionNameFromActivityInvocation(InvocationExpressionSyntax
         }
 
-        private bool TryGetInputNodeFromCallActivityInvocation(InvocationExpressionSyntax invocationExpression, out SyntaxNode inputNode)
+        private static bool TryGetInputNodeFromCallActivityInvocation(SemanticModel semanticModel, InvocationExpressionSyntax invocationExpression, out SyntaxNode inputNode)
+        {
+            // If CallActivity is a custom method in user code
+            if (SyntaxNodeUtils.TryGetDeclaredSyntaxNode(semanticModel, invocationExpression, out SyntaxNode declaration))
+            {
+                if (TryGetSpecificParameterIndex(declaration, "object input", out int inputParameterIndex))
+                {
+                    if (TryGetInvocationArguments(invocationExpression, out IEnumerable<ArgumentSyntax> arguments))
+                    {
+                        var argumentNode = arguments.ElementAt(inputParameterIndex);
+                        inputNode = argumentNode.ChildNodes().First();
+                        return true;
+                    }
+                }
+            }
+            // else assume CallActivity is a DurableFunctions method
+            else
+            {
+                if (TryGetInvocationArguments(invocationExpression, out IEnumerable<ArgumentSyntax> arguments))
+                {
+                    // Input node is currently the last argument on CallActivity* methods. If this is changed, this will not be sufficient to 
+                    // determine which argument is meant to represent the input.
+                    var argumentNode = arguments.Last();
+                    inputNode = argumentNode.ChildNodes().First();
+                    return true;
+                }
+            }
+
+            inputNode = null;
+            return false;
+        }
+
+        private static bool TryGetSpecificParameterIndex(SyntaxNode declaration, string parameterToFind, out int inputParameterIndex)
+        {
+            if (declaration is MethodDeclarationSyntax methodDeclaration)
+            {
+                var parameters = methodDeclaration.ParameterList.ChildNodes();
+                var length = parameters.Count();
+                for (int i = 0; i < length; i++)
+                {
+                    if (parameters.ElementAt(i).ToString() == parameterToFind)
+                    {
+                        inputParameterIndex = i;
+
+                        if (IsExtensionMethod(parameters))
+                        {
+                            inputParameterIndex--;
+                        }
+
+                        return true;
+                    }
+                }
+            }
+
+            inputParameterIndex = int.MinValue;
+            return false;
+        }
+
+        private static bool IsExtensionMethod(IEnumerable<SyntaxNode> parameters)
+        {
+            var firstParameter = parameters.ElementAt(0);
+            if (firstParameter.ToString() == "this IDurableOrchestrationContext context")
```

Comment:
What if the developer decides to name the parameter something other than "context" (like "ctx" or "dfContext")? Maybe we could instead do something like this?
```suggestion
            if (firstParameter.ToString().StartsWith("this IDurableOrchestrationContext "))
```

Reply:
Good point.